test/shopify_app/controller_concerns/login_protection_test.rb

```diff
@@ -63,6 +65,69 @@ class LoginProtectionTest < ActionController::TestCase
     end
   end
 
+  test "#current_shopify_session retrieves user session using jwt" do
+    ShopifyApp.configuration.allow_jwt_authentication = true
```

Comment:
Do you want to reset this to `false` after each test runs? You could do this in an `after` block.